src/concurrency.h

```diff
@@ -26,9 +26,10 @@ class single_consumer_queue
     // when need to stop
     std::atomic<bool> _need_to_flush;
     std::atomic<bool> _was_flushed;
+    std::function<void(T)> _print_on_drop;
 public:
-    explicit single_consumer_queue<T>(unsigned int cap = QUEUE_MAX_SIZE)
-        : _queue(), _mutex(), _deq_cv(), _enq_cv(), _cap(cap), _accepting(true), _need_to_flush(false), _was_flushed(false)
+    explicit single_consumer_queue<T>(unsigned int cap = QUEUE_MAX_SIZE, std::function<void(T)> print_on_drop = nullptr)
```

Comment:
`on_drop_callback`

Reply:
the function takes T? why not `T const &`?

Reply:
done